packages/atomic/src/utils/object-utils.tsx

```diff
@@ -0,0 +1,17 @@
+export function readFromObject<T extends object>(
+  object: T,
+  key: string
+): string | undefined {
+  const keys = key.split('.');
+  let current: unknown = object;
+
+  for (const k of keys) {
+    if (current && typeof current === 'object' && k in current) {
+      current = (current as Record<string, unknown>)[k];
+    } else {
+      return undefined;
+    }
+  }
+
+  return typeof current === 'string' ? current : undefined;
+}
```

Comment:
It was actually bugged in commerce because the readFromObject function was different than the one in search. I made a common function for the two of them here with the one from search.